### PR TITLE
Add FetchMessagesServlet and use SendTextMessageServlet instead of directly calling firebase from js

### PR DESCRIPTION
**Pull request by arjavibahety**
- [X] Move firebase code to get messages to backend
- [X] Use text messages servlet and prevent chat box from reloading
- [X] DI for fetch messages servlet
- [X] DI for textmessages servlet

### CODE REVIEW

File: webproject/src/main/java/com/google/sps/dataManagers/MessagesManager.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        if (!userRoomManager.isRoomIdValid(roomId)) {
+            throw new ServletException("Invalid roomId");
+        }
+
+        DatabaseReference ref = firebaseUtil.getMessagesReference().child(roomId);
+        List<DataSnapshot> dataSnapshots = firebaseUtil.getAllSnapshotsFromReference(ref);
+
+        List<Message> messages = new ArrayList<>();
+        for (DataSnapshot dataSnapshot : dataSnapshots) {
+            String user = dataSnapshot.child("user").getValue(String.class);
```
Comment (kevinluvian) [inline webproject/src/main/java/com/google/sps/dataManagers/MessagesManager.java]: move to message class

**Review by kevinluvian — CHANGES_REQUESTED** [review]

File: webproject/src/main/java/com/google/sps/dataManagers/MessagesManager.java (hunk 1 above)
Comment (arjavibahety) [reply webproject/src/main/java/com/google/sps/dataManagers/MessagesManager.java]: changed it to use stream to get the list of messages

Comment (arjavibahety) [reply webproject/src/main/java/com/google/sps/dataManagers/MessagesManager.java]: changed it to use stream to get a list of messages

File: webproject/src/test/java/com/google/sps/authentication/AuthenticationHandlerTest.java [hunk 2]
```diff
@@ -44,7 +44,7 @@ public void isUserLoggedIn_notLoggedIn_returnsFalse() {
         boolean isUserLoggedIn = authenticationHandler.isUserLoggedIn();
 
         // Assert
-        assertTrue(isUserLoggedIn);
+        assertFalse(isUserLoggedIn);
```
Comment (kevinluvian) [inline webproject/src/test/java/com/google/sps/authentication/AuthenticationHandlerTest.java:47]: why assert got changed?

**Review by kevinluvian — APPROVED** [review]

**Pull request merged**